pkgs/by-name/po/porn-vault/package.nix

```diff
@@ -0,0 +1,103 @@
+{
+  fetchFromGitLab,
+  fetchurl,
+  rustPlatform,
+  lib,
+  pnpm,
+  stdenvNoCC,
+  nodejs_22,
+  ffmpeg,
+  imagemagick,
+  writeShellScript,
+  makeWrapper,
+  autoPatchelfHook,
+}:
+let
+  izzy = rustPlatform.buildRustPackage rec {
+    pname = "izzy";
+    version = "2.0.1";
+
+    src = fetchFromGitLab {
+      owner = "porn-vault";
+      repo = "izzy";
+      rev = version;
+      hash = "sha256-UauA5mZi5a5QF7d17pKSzvyaWbeSuFjBrXEAxR3wNkk=";
+    };
+
+    cargoHash = "sha256-y0bV83bnuSALwvRD8ztOiBE6LSg3Hx6HKSjvQPEkvIQ=";
+    cargoPatches = [ ./add-Cargo.lock.patch ];
+
+    meta = with lib; {
+      description = "Rust In-Memory K-V Store with Redis-Style File Persistence and Secondary Indices";
+      homepage = "https://gitlab.com/porn-vault/izzy";
+      license = lib.licenses.gpl3;
+      maintainers = with maintainers; [ luNeder ];
+      platforms = lib.platforms.all;
+      mainProgram = "izzy";
+    };
+  };
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "porn-vault";
+  version = "0.30.0-rc.11";
+
+  src = fetchFromGitLab {
+    owner = "porn-vault";
+    repo = "porn-vault";
+    rev = "4c6182c5825d85193cf67cb7cd927da2feaaecdb";
+    hash = "sha256-wQ3dqLc0l2BmLGDYrbWxX2mPwO/Tqz0fY/fOQTEUv24=";
+  };
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-Xr9tRiP1hW+aFs9FnPvPkeJ0/LtJI57cjWY5bZQaRTQ=";
+  };
+
+  nativeBuildInputs = [
+    nodejs_22
+    pnpm.configHook
+    makeWrapper
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    pnpm build
+    sed s+%out%+$out+ ${./porn-vault.sh} > ./porn-vault.sh
+    sed s+%nodejs%+${nodejs_22}+ -i ./porn-vault.sh
+    sed s+%izzy%+${izzy}+ -i ./porn-vault.sh
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 porn-vault.sh "$out/bin/porn-vault"
+    install -Dm644 package.json config.example.json remix.config.js -t $out/share/porn-vault
+    cp -R public dist build node_modules graphql locale -t $out/share/porn-vault
+    install -Dm111 ${izzy}/bin/izzy "$out/share/porn-vault/izzy"
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    wrapProgram "$out/bin/porn-vault" \
+      --prefix PATH : "${
+        lib.makeBinPath [
+          ffmpeg
+          imagemagick
+          izzy
+        ]
+      }"
+  '';
+
+  meta = {
+    description = "Porn-Vault is a self hosted organizer for adult videos and imagery.";
+    homepage = "https://gitlab.com/porn-vault/porn-vault";
+    license = lib.licenses.gpl3;
```

Comment:
gpl3 is deprecated and ambiguous: use either gpl3Only or gpl3Plus - if it's not explicitly "or later" then go with gpl3Only